src/Particle/HDFWalkerOutput.cpp

```diff
@@ -93,22 +93,41 @@ bool HDFWalkerOutput::dump(const WalkerConfigurations& W, int nblock)
 
   //try to use collective
   hdf_archive dump_file(myComm, true);
-  dump_file.create(FileName);
-  HDFVersion cur_version;
-  dump_file.write(cur_version.version, hdf::version);
+  bool exists = dump_file.open(FileName);
+  if (!exists) // create new config.h5
+  {
+    dump_file.create(FileName);
+    HDFVersion cur_version;
+    dump_file.write(cur_version.version, hdf::version);
+  }
   dump_file.push(hdf::main_state);
-  dump_file.write(nblock, "block");
 
-  write_configuration(W, dump_file, nblock);
+  write_configuration(W, dump_file, nblock, identify_block);
   dump_file.close();
 
   currentConfigNumber++;
   prevFile = FileName;
   return true;
 }
 
-void HDFWalkerOutput::write_configuration(const WalkerConfigurations& W, hdf_archive& hout, int nblock)
+void HDFWalkerOutput::write_configuration(const WalkerConfigurations& W, hdf_archive& hout, int nblock, const bool identify_block)
 {
+  std::string partition_name = "walker_partition";
+  std::string dataset_name = hdf::walkers;
+  std::string weights_name = hdf::walker_weights;
+  if (identify_block)
+  { // change h5 slab name to record more than one block
+    std::stringstream block_str;
+    block_str << nblock;
+    partition_name += block_str.str();
+    dataset_name += block_str.str();
+    weights_name += block_str.str();
+  } else { // remove previous checkpoint
+    std::vector<std::string> names = {"block", hdf::num_walkers, partition_name, dataset_name, weights_name};
+    for (auto aname : names)
+      if (hout.is_dataset(aname)) hout.unlink(aname);
+  }
```

Comment:
could you mentioned the data set name used for recording instead of checkpointing.

Reply:
Dataset name for recording is `walkers{block_idx}`, whereas the name for checkpoint is simply `walkers`.